src/test/TestUtils/TestProjectFixture.cs

```diff
@@ -277,16 +277,23 @@ private void ValidateRequiredDirectories(RepoDirectoriesProvider repoDirectories
             return this;
         }
 
-        public TestProjectFixture RestoreProject(string[] fallbackSources)
+        public TestProjectFixture RestoreProject(string[] fallbackSources, string extraMSBuildProperties = null)
         {
             var restoreArgs = new List<string>();
             foreach (var fallbackSource in fallbackSources)
             {
-                restoreArgs.Add("-f");
+                restoreArgs.Add("--source");
                 restoreArgs.Add(fallbackSource);
             }
             restoreArgs.Add("--disable-parallel");
 
+            restoreArgs.Add($"/p:MNAVersion={_repoDirectoriesProvider.MicrosoftNETCoreAppVersion}");
```

Comment:
I don't get it.  The project doesn't need to define `RuntimeFrameworkVersion` at all with my proposal.

When you are saying "a slight preference for this", does "this" mean what I am proposing?  or what the code is doing now?